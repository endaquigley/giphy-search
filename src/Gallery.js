import React from "react";
import styled from "styled-components";
import { useSelector, useDispatch } from "react-redux";

import { useFetchImages } from "./hooks";
import Thumbnail from "./Thumbnail.js";
import * as actions from "./actions";

const Wrapper = styled.div`
  margin: 15px;
  display: grid;
  grid-gap: 15px;
  --thumbnail-width: 85px;
  grid-template-columns: repeat(auto-fit, minmax(var(--thumbnail-width), 1fr));

  @media (min-width: 690px) {
    --thumbnail-width: 120px;
  }
`;

export const Gallery = React.memo(() => {
  const dispatch = useDispatch();
  const data = useSelector(state => state.data);

  useFetchImages();

  const updateSelected = selected => {
    return dispatch(actions.updateSelected(selected));
  };

  return (
    <Wrapper>
      {data.map(media =>
        media.images.preview_gif === undefined ? null : (
          <Thumbnail
            key={media.id}
            source={media.images.preview_gif.url}
            handleClick={() => updateSelected(media)}
          />
        )
      )}
    </Wrapper>
  );
});
