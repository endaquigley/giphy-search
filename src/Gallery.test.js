import React from 'react';
import ReactDOM from 'react-dom';
import Gallery from './Gallery';

it('renders no video elements when data is empty', () => {
  const component = document.createElement('div');
  ReactDOM.render(<Gallery data={ [] } />, component);

  const videos = component.querySelectorAll('video');
  expect(videos.length).toBe(0);

  expect(component).toMatchSnapshot();
});

it('renders a video element for each element in the data array', () => {
  const data = [
    {
      id: 1,
      images: {
        preview: { mp4: 'https://enda.ie/filename-01.mp4' },
        original: { mp4: 'https://enda.ie/filename-01.mp4' }
      }
    },
    {
      id: 2,
      images: {
        preview: { mp4: 'https://enda.ie/filename-02.mp4' },
        original: { mp4: 'https://enda.ie/filename-02.mp4' }
      }
    }
  ];

  const component = document.createElement('div');
  ReactDOM.render(<Gallery data={ data } />, component);

  const videos = component.querySelectorAll('video');
  expect(videos.length).toBe(2);
  
  expect(videos[0].src).toBe('https://enda.ie/filename-01.mp4');
  expect(videos[1].src).toBe('https://enda.ie/filename-02.mp4');

  expect(component).toMatchSnapshot();
});