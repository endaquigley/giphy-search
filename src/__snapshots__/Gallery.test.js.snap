// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an image element for each element in the data array 1`] = `
<div>
  <div
    class="sc-bZQynM jgZutv"
  >
    <div
      class="sc-bdVaJa kiUbly"
    >
      <div
        class="sc-bwzfXH fFwymd"
      >
        <img
          class="sc-htpNat jtiZZo"
          src="https://enda.ie/filename-01.gif"
        />
      </div>
    </div>
    <div
      class="sc-bdVaJa kiUbly"
    >
      <div
        class="sc-bwzfXH fFwymd"
      >
        <img
          class="sc-htpNat jtiZZo"
          src="https://enda.ie/filename-02.gif"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders no image elements when data is empty 1`] = `
<div>
  <div
    class="sc-bZQynM jgZutv"
  />
</div>
`;
